core-rust/state-manager/src/jni/node_rust_environment.rs

```diff
@@ -141,10 +146,18 @@ impl JNINodeRustEnvironment {
             .track_running_tasks()
             .measured(metric_registry.deref());
 
+        let protocol_updater_factory: Box<dyn ProtocolUpdaterFactory + Send + Sync> =
```

Comment:
Probably cleaner if we are more explicit about what `ProtocolUpdaterFactory` is and implement it as an enum rather than dynamic trait.